Ligandr/LigOverlay.py

```diff
@@ -0,0 +1,217 @@
+"""
+Places a ligand or multiple ligands in place of another ligand on a protein
+"""
+import sys
+import tempfile
+import traceback
+from typing import List
+
+import NameToSMILES
+
+import mdtraj
+import numpy as np
+from mdtraj import Trajectory
+from openeye import oechem, oedocking, oeomega, oeff
+from openeye.oechem import OEMol, OEGraphMol
+import itertools
+import subprocess
+
+
+def dock_molecule(trajectory_frame, original_ligands, new_ligands, title_list, keep_ions=None,
+                  override_replacement=False) -> List[
+    mdtraj.Trajectory]:
+    """
+    Docks ligands or a list of ligands in place of another ligand
+
+    Parameters
+    ----------
+    trajectory_frame: mdtraj.Trajectory, required
+        The trajectory to dock to- water will be removed, but every other atoms will stay the same
+
+    original_ligands: [str], required
+        The original ligand to replace, must be an mdtraj selection string. If more than
+        one residue or no residue is selected, program will throw an index error
+
+    new_ligands: [[str]], required
+        For each original ligand, a list of SMILES string representing the new ligand to dock.
+
+    title_list: [[str]], required
+        For each new ligand, the title the user wishes to have.
+
+    keep_ions: [str], optional, default:None
+        A list of ions to keep when removing solvent. Must be an mdtraj selection.
+
+    override_replacement: bool, optional, default:False
+        Setting to true will prevent replacing ligand with canonical molecule from PDB.
+
+    Returns
+    -------
+    output: [mdtraj.Trajectory]
+        A list of trajectories for all combinations of ligands
+    """
+    # Check inputs
+    if len(original_ligands) != len(new_ligands):
+        raise ValueError("Original and new ligand arrays must have the same length.")
+    # Produces pdb file in memory
+    overall_dry_frame = trajectory_frame.remove_solvent(exclude=keep_ions)  # type: mdtraj.Trajectory
+    complex_prot = __trajectory_to_oemol__(overall_dry_frame)
+
+    # Bookkeeping
+    ligand_stream = []
+    output = []
+    i = 0
+    # Being processing a ligand
+    for ligand in original_ligands:
+        try:
+            ligand_traj = overall_dry_frame.atom_slice(overall_dry_frame.top.select(ligand))  # type: mdtraj.Trajectory
+        except IndexError as e:
+            print("Ligand selection was invalid")
+            print("Exception: %s" % str(e), file=sys.stderr)
+            traceback.print_tb(e.__traceback__, file=sys.stderr)
+            sys.exit(1)
+        if ligand_traj.n_residues > 1:
+            raise IndexError("More than one ligand residue found. Only one expected")
+        # Find approximate location of binding pocket
+        ligand_center = np.average(ligand_traj.xyz[0], axis=0) * 10
```

Comment:
Could you please add a comment line about the `* 10` here? Is this a unit conversion for something later that assumes `ligand_center` is in a different unit than `ligand_traj.xyz`?